hazelcast/src/main/java/com/hazelcast/spi/impl/BasicInvocationFuture.java

```diff
@@ -283,7 +287,7 @@ private static long getCallTimeoutExtension(long callTimeout) {
     }
 
     int getWaitingThreadsCount() {
-        return waiterCount.get();
+        return WAITER_COUNT_FIELD_UPDATER.get(this);
```

Comment:
I could even simplify that with a return waiterCount; no need to go through the field updater.